src/Microsoft.Identity.Web/AuthorityHelpers.cs

```diff
@@ -6,35 +6,35 @@
 
 namespace Microsoft.Identity.Web
 {
-    internal static class AuthorityHelpers
-    {
-        internal static string BuildAuthority(MicrosoftIdentityOptions options)
-        {
-            Uri baseUri = new Uri(options.Instance);
-            string pathBase = baseUri.PathAndQuery.TrimEnd('/');
-            var domain = options.Domain;
-            var tenantId = options.TenantId;
+	internal static class AuthorityHelpers
+	{
+		internal static string BuildAuthority(MicrosoftIdentityOptions options)
+		{
+			Uri baseUri = new Uri(options.Instance);
+			string pathBase = baseUri.PathAndQuery.TrimEnd('/');
+			var domain = options.Domain;
+			var tenantId = options.TenantId;
 
-            if (options.IsB2C)
-            {
-                var userFlow = options.DefaultUserFlow;
-                return new Uri(baseUri, new PathString($"{pathBase}/{domain}/{userFlow}/v2.0")).ToString();
-            }
-            else
-            {
-                return new Uri(baseUri, new PathString($"{pathBase}/{tenantId}/v2.0")).ToString();
-            }
-        }
+			if (options.IsB2C)
+			{
+				var userFlow = options.DefaultUserFlow;
+				return new Uri(baseUri, new PathString($"{pathBase}/{domain}/{userFlow}/v2.0")).ToString();
+			}
+			else
+			{
+				return new Uri(baseUri, new PathString($"{pathBase}/{tenantId}/v2.0")).ToString();
+			}
+		}
 
-        internal static string EnsureAuthorityIsV2(string authority)
-        {
-            authority = authority.Trim().TrimEnd('/');
-            if (!authority.EndsWith("v2.0", StringComparison.InvariantCulture))
-            {
-                authority += "/v2.0";
-            }
+		internal static string EnsureAuthorityIsV2(string authority)
+		{
+			authority = authority.Trim().TrimEnd('/');
```

Comment:
String allocations tend to be a death by 1000 cuts issue at scale. Instead of TrimEnd('/'), I suggest adding "v2.0" or "/v2.0" based upon the presence of the trailing slash. Doing so should skip one of the string reallocations.